backend/endpoints/files.py

```diff
@@ -0,0 +1,98 @@
+from fastapi import  HTTPException, status
+from schema.response import ResponseModel
+from starlette.responses import JSONResponse, Response
+from fastapi import FastAPI, UploadFile, File, APIRouter
+from fastapi.responses import FileResponse, JSONResponse
+from utils.message_utils import get_message, get_room_messages
+import os
+import zipfile
+import io
+from zipfile import ZipFile
+router = APIRouter()
+
+"""
+ A get method endpoint to Handle retrieving and Viewing of uploaded files in a DM
+
+"""
+
+# {
+#     "organization_id" :"6373eb474746182adae97314",
+#     "room_id" : "6373eb4f4746182adae97316"
+
+        #         org_id:
+        # 6373eb474746182adae97314
+        # room_id:
+        # 6373eb4f4746182adae97316
+
+        #  Use:
+        # org_id: 619ba4
+        # room_id: 123456
+# }
+
+
+def zipfiles(filenames: str):
+    zip_subdir = "dummy_archive_path"
+    zip_filename = "archive.zip"
+    """
+        filenames = ["dummy_url path1", "dummy_url path12" ]
+    """
+    data = io.BytesIO()
+    temp = zipfile.ZipFile(data, "w")
+
+    for filepath in filenames:
+        fdir, fname = os.path.split(filepath)
+        zip_path = os.path.join(zip_subdir, fname)
+        temp.write(fdir, zip_path)
+
+    temp.close()
+    resp = Response(data.getvalue(), media_type="image/png-compressed", headers={
+        'Content-Disposition': f'attachment;filename={zip_filename}'
+    })
+
+    return response
```

Comment:
how did you manage to test this since `response` is not defined anywhere?

Reply:
return response was a mistake while trying to clean up code to improve readability.. now rectified
endpoint retested and works properly like in my screenshots shared in previous comments
i have also corrected placement of the docstrings; placed them inside the function
i checked other endpoints on the codebase  and saw that it followed this endpoint route patern : "/org/{org_id}/rooms/{room_id}/files"  as this was also suggested to be used by a previous second reviewer
i also referred to other endpoint routes pattern eg; the get message endpoint route pattern "/org/{org_id}/rooms/{room_id}/messages"